runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -954,6 +963,41 @@ private <K, V> void redistributeByKeyHelper(
           }
         });
 
+    registerTransformTranslator(
+        RedistributeArbitrarily.class,
+        new TransformTranslator<RedistributeArbitrarily>() {
+          @Override
+          public void translate(RedistributeArbitrarily transform, TranslationContext context) {
+            redistributeArbitrarilyHelper(transform, context);
+          }
+
+          private <T> void redistributeArbitrarilyHelper(
+              RedistributeArbitrarily<T> transform, TranslationContext context) {
+            StepTranslationContext stepContext = context.addStep(transform, "GroupByKey");
```

Comment:
This is a problem because the input is not necessarily keyed. I think the change you made above should make the RedistributeByKey translator suffice